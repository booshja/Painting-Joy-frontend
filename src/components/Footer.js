// dependencies
import React from "react";
import styled from "styled-components";
// components
import { Link } from "react-router-dom";
import TitleLogoFooter from "./TitleLogoFooter";
import NavFooter from "./NavFooter";
// breakpoints
import { breakpoints } from "../breakpoints";

const StyledFooter = styled.footer`
    background-color: #012e34;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-around;
    color: #ffffff;
    padding: 1rem;
    min-height: 150px;
    margin-top: 2rem;
`;

const StyledTitle = styled(Link)`
    font-family: "Barlow Condensed", sans-serif;
    font-weight: 700;
    font-size: 1.4rem;
    letter-spacing: 1px;

    ${breakpoints("font-size", "rem", [{ 414: 1.8 }])}
    ${breakpoints("display", "", [{ 1024: "none" }])}
`;

const StyledLink = styled(Link)`
    font-family: "News Cycle", sans-serif;
    font-size: 1.1rem;
    letter-spacing: 1px;

    ${breakpoints("font-size", "rem", [{ 414: 1.4 }])}
    ${breakpoints("display", "", [{ 1024: "none" }])}
`;

const StyledCopyright = styled.p`
    font-family: "News Cycle", sans-serif;
    font-size: 0.6rem;
    letter-spacing: 1px;

    ${breakpoints("font-size", "rem", [{ 414: 0.8 }])}
`;

const StyledTitleLogo = styled(TitleLogoFooter)``;

const StyledNavFooter = styled(NavFooter)``;

const StyledContainer = styled.div`
    margin: 1rem 0 2rem 0;

    ${breakpoints("display", "", [{ 360: "none" }, { 1024: "flex" }])}
    ${breakpoints("justify-content", "", [{ 1024: "space-between" }])}
    ${breakpoints("width", "%", [{ 1024: 60 }])}
`;

const Footer = () => {
    return (
        <StyledFooter>
            <StyledTitle to="/">Painting Joy Mural Co.</StyledTitle>
            <StyledLink to="/contact">Contact Me</StyledLink>
            <StyledContainer>
                <StyledTitleLogo />
                <StyledNavFooter />
            </StyledContainer>
            <StyledCopyright>
                Copyright &copy; 2021 - Painting Joy Mural Co.
            </StyledCopyright>
        </StyledFooter>
    );
};

export default Footer;
