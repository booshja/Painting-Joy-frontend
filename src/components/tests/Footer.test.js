import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Footer from "../Footer";

/** Smoke Test */
it("renders without crashing", () => {
    render(
        <BrowserRouter>
            <Footer />
        </BrowserRouter>
    );
});

/** Snapshot Test */
it("matches snapshot", () => {
    const { asFragment } = render(
        <BrowserRouter>
            <Footer />
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

/** RTL Tests */
it("renders correctly", () => {
    const { getByText, getByAltText } = render(
        <BrowserRouter>
            <Footer />
        </BrowserRouter>
    );

    getByText("Painting Joy Mural Co.");
    getByText("Contact Me");
    getByText("2021 - Painting Joy Mural Co", { exact: false });
    // children
    // TitleLogoFooter Component
    getByAltText("Painting Joy Mural Co. Logo");
    // NavFooter Component
    getByText("Home");
    getByText("Murals");
    getByText("Contact");
    getByText("Store");
});
