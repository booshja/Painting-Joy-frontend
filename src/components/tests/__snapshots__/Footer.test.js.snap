// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <footer
    class="sc-jSFjdj ijiXtL"
  >
    <a
      class="sc-gKAaRy beJoPo"
      href="/"
    >
      Painting Joy Mural Co.
    </a>
    <a
      class="sc-iCoGMd cZLOhu"
      href="/contact"
    >
      Contact Me
    </a>
    <div
      class="sc-pNWdM cIWBiw"
    >
      <div
        class="sc-bdnxRM crSiVT"
      >
        <a
          href="/"
        >
          <img
            alt="Painting Joy Mural Co. Logo"
            class="sc-dlnjwi hEOZFd"
            src="painting-joy-logo-white.jpg"
          />
        </a>
      </div>
      <div
        class="sc-hKFxyN fGvBCY"
      >
        <a
          class="sc-eCApnc cFONQn"
          href="/"
        >
          Home
        </a>
        <a
          class="sc-eCApnc cFONQn"
          href="/murals"
        >
          Murals
        </a>
        <a
          class="sc-eCApnc cFONQn"
          href="/contact"
        >
          Contact
        </a>
        <a
          class="sc-eCApnc cFONQn"
          href="/store"
        >
          Store
        </a>
      </div>
    </div>
    <p
      class="sc-fujyAs jfwRNI"
    >
      Copyright Â© 2021 - Painting Joy Mural Co.
    </p>
  </footer>
</DocumentFragment>
`;
